### PR TITLE
Add: Testing best practices

### PR DESCRIPTION
**Pull request by dinhtungdu**
### Description of the Change
This PR:
- Adds a new Testing page to 10up Open Source Best Practice.
- Moves the Code Coverage section over.
- Adds E2E testing section to the new Testing page.

### Alternate Designs

I wanted to have a detailed walkthrough/instruction on how to set up Cypress for E2E testing. But a tutorial doesn't fit the writing style of Open Source Best Practices.

### CODE REVIEW

File: _includes/markdown/Testing.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+Every feature should be accompanied by tests and all pull requests should come with associated tests, all living within the `tests` directory.  While we have no intention of striving for 100% code coverage, we should aim for above 80% with above 90% being the ideal.  We should also look to utilize a code coverage / automated code review tool like [Coveralls](https://coveralls.io/), [Code Climate](https://codeclimate.com/), or [Codecov](https://codecov.io/) and ensure that is a pull request requirement before merging.
+
+<!-- @todo: add coverage badge details -->
+
+<h2 id="e2e-testing" class="anchor-heading">E2E Testing {% include Util/link_anchor anchor="e2e-testing" %} {% include Util/top %}</h2>
+
+### Testing environment
+
+The environment should be accessible and easy to set up in both local and CI environments. In the WordPress context, the environment must also be configurable to fit as many scenarios as possible, including testing against different WP versions and with other plugins/themes activated. For the open-source projects, we also want to use a community-driven tool. Because of those reasons, we choose `wp-env` for 10up's Open Source projects.
```
Comment (jeffpaul) [inline _includes/markdown/Testing.md]: Let's add a link out to the best resource on what `wp-env` is on that reference here.

File: _includes/markdown/Testing.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+### Test runner
+
+Cypress is the tool we're using for E2E testing. Cypress provides a great developer experience for writing the tests, including simple and familiar syntax and a powerful visual debugger. On top of that, Cypress gives us the most consistent results among the tools we tried.
```
Comment (jeffpaul) [inline _includes/markdown/Testing.md]: Please add a link out for reference on what Cypress is via this first reference here.

**Comment by jeffpaul** [comment]
@Sidsector9 would be good to get your review on this write-up to see if there's anything else that would be helpful to call out here that was helpful for you in implementing this on Restricted Site Access.

**Review by Sidsector9 — APPROVED** [review]
This all looks good @jeffpaul. I think Tung has covered almost everything in this PR. 
On another note, is it possible to add few GIFs of Cypress in action on Simple Podcasting? I think it will be eye-catchy for the article.

**Comment by jeffpaul** [comment]
@dinhtungdu I think a couple GIF visuals will help as @Sidsector9 mentioned.  I'll let you two sort out which parts of the e2e flow would be best to have in GIF format and then get those added in (could be a subsequent PR if it'll take a while).

**Pull request merged**